src/components/Navbar/hooks.ts

```diff
@@ -0,0 +1,18 @@
+import { useRef } from "react";
+
+type UseDotHookProps = {
+  items: number;
+  activeId?: number;
+};
+
+export default function useDotAnimation({ items, activeId }: UseDotHookProps) {
+  const spaceRef = useRef<HTMLDivElement>(null);
+  const listRef = useRef<HTMLUListElement>(null);
+
+  listRef.current?.style.setProperty("--navbar-items", "" + items);
```

Comment:
bug/todo: Let's avoid using the direct DOM ref to set the class. Instead, it should be applied in the virtual DOM, as you had before. 

This hook only needs to return `{ style }` with all the variables, and then you pass them the `<ul style={style} />` the vars there automatically work in any child element.

You don't even need useRef here. Gotcha?

Reply:
Gotcha 🫡 funny thing I had to cast the object with `as React.CSSProperties` otherwise TS would keep complaining.
This refactor is to reduce complexity, right? It has no effect on performance? 🤔

Reply:
Exactly, don't expect any downside with performance. And yes, `React.CSSProperties` is great to ensure you pass valid CSS properties!

TS Question: Why do you need `String()` in `"--navbar-items": String(items)`? 🤔

Reply:
I don't haha. Updated (maybe it was before doing the as React.CSSProperties)